// kaped_partner_api/src/model/Business.js
/*
 * KAPED PARTNER API
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: 1.2.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 3.0.37
 *
 * Do not edit the class manually.
 *
 */
import {ApiClient} from '../ApiClient';

/**
 * Enum class Business.
 * @enum {String}
 * @readonly
 */
const Business = {
  /**
   * value: "LLC"
   * @const
   */
  LLC: "LLC",

  /**
   * value: "Corporation"
   * @const
   */
  corporation: "Corporation",

  /**
   * value: "Non Profit"
   * @const
   */
  nonProfit: "Non Profit",

  /**
   * value: "Sole Proprietor"
   * @const
   */
  soleProprietor: "Sole Proprietor",

  /**
   * value: "Government"
   * @const
   */
  government: "Government",

  /**
   * Returns a <code>Business</code> enum value from a JavaScript object name.
   * @param {Object} data The plain JavaScript object containing the name of the enum value.
  * @return {module:model/Business} The enum <code>Business</code> value.
   */
  constructFromObject: function(object) {
    return object;
  }
};

export {Business};
